awx/api/views/analytics.py

```diff
@@ -0,0 +1,274 @@
+            text["error"]["message"] = message
+        return Response(text, status=status_code)
+
+    def _error_response_404(self, response):
+        try:
+            json_response = response.json()
+            # Subscription/entitlement problem or missing tenant data in AA db => HTTP 403
+            message = json_response.get('error', None)
+            if message:
+                return self._error_response(ERROR_NO_DATA_OR_ENTITLEMENT, message, remote=True, remote_status_code=response.status_code)
+
+            # Standard 404 problem => HTTP 404
+            message = json_response.get('detail', None) or response.text
+        except requests.exceptions.JSONDecodeError:
+            # Unexpected text => still HTTP 404
+            message = response.text
+
+        return self._error_response(ERROR_NOT_FOUND, message, remote=True, remote_status_code=status.HTTP_404_NOT_FOUND, status_code=status.HTTP_404_NOT_FOUND)
+
+    @staticmethod
+    def _forward_response(response):
+        try:
+            content_type = response.headers['content-type'] or ''
+            if content_type.find('application/json') != -1:
+                return Response(response.json(), status=response.status_code)
+        except Exception as e:
+            logger.error(f"Analytics API: Response error: {e}")
+
+        return Response(response.content, status=response.status_code)
+
+    def _send_to_analytics(self, request, method):
+        try:
+            headers = self._request_headers(request)
+
+            self._check_upload_enabled()
+            url = self._get_analytics_url(request.path)
+            rh_user = self._get_rh_user()
+            rh_password = self._get_rh_password()
+
+            if method == "GET":
+                response = requests.get(url, auth=(rh_user, rh_password), verify=CERT_PATH, params=request.query_params, headers=headers, json=request.data)
+            elif method == "POST":
+                response = requests.post(url, auth=(rh_user, rh_password), verify=CERT_PATH, params=request.query_params, headers=headers, json=request.data)
+            else:
+                return self._error_response(ERROR_UNSUPPORTED_METHOD, method, remote=False, status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+            #
+            # Missing or wrong user/pass
+            #
+            if response.status_code == status.HTTP_401_UNAUTHORIZED:
+                text = (response.text or '').rstrip("\n")
+                return self._error_response(ERROR_UNAUTHORIZED, text, remote=True, remote_status_code=response.status_code)
+            #
+            # Not found, No entitlement or No data in Analytics
+            #
+            elif response.status_code == status.HTTP_404_NOT_FOUND:
+                return self._error_response_404(response)
+            #
+            # Success or not a 401/404 errors are just forwarded
+            #
+            else:
+                return self._forward_response(response)
```

Comment:
checked: redirection is handled by default for GET and POST (https://requests.readthedocs.io/en/latest/user/quickstart/#redirection-and-history).  

In case of request to `'https://localhost:8043/api/v2/analytics/report/most_used_modules'` (Controller's URL without slash), it returns 301 with correct AWX location. 

**Tested.**

Reply:
Perfect, thanks!